src/main/java/hudson/plugins/sauce_ondemand/SauceEnvironmentUtil.java

```diff
@@ -209,4 +212,10 @@ public static String getBuildName(AbstractBuild<?, ?> build) {
         }
         return builder.toString();
     }
+
+    public static String generateTunnelIdentifier(final AbstractBuild build) {
```

Comment:
how about changing this function signature to 
    public static String generateTunnelIdentifier(final String projectName)
?
then unit tests that I see in this pull request will be dramatically simplified.
what this method does is essentially "get a string and produce another string". having a dependency on Jenkins model element AbstractBuild is not necessary  and forces to add mocks. I will add a comment to the tests section.